File: app/src/main/java/com/app/cheffyuser/cart/adapter/CustomOrderAdapter.java
package com.app.cheffyuser.cart.adapter;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.app.cheffyuser.R;
import com.app.cheffyuser.cart.activities.CustomOrderListActivity;

public class CustomOrderAdapter extends RecyclerView.Adapter<CustomOrderAdapter.MyViewHolder> {


    Context context;
    private String foodItemList[];
    private int imgList[];

    //constructor
    public CustomOrderAdapter(Context context, String foodItemList[], double foodPriceList[], int imgList[]) {
        this.context = context;
        this.foodItemList = foodItemList;
        this.imgList=imgList;
    }



    @NonNull
    public CustomOrderAdapter.MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType)
    {

        // infalte the item Layout
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.single_custom_order_row, parent, false);
        // set the view's size, margins, paddings and layout parameters
       CustomOrderAdapter.MyViewHolder vh = new CustomOrderAdapter.MyViewHolder(view); // pass the view to View Holder
        return vh;
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder myHolder, final int position) {



        myHolder.txtItemName.setText(foodItemList[position]);

        //myHolder.txtI.setText(foodItemList[position]);



//        myHolder.imgFood.setImageResource(imgList[position]);
        // implement setOnClickListener event on item view.
        myHolder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                // display a toast with person name on item click
                Toast.makeText(context, foodItemList[position], Toast.LENGTH_SHORT).show();

                Intent intent=new Intent(context, CustomOrderListActivity.class);
                context.startActivity(intent);
            }
        });
    }


    //for item count
    @Override
    public int getItemCount() {
        return foodItemList.length;
    }


    //create class
    public class MyViewHolder extends RecyclerView.ViewHolder{

        TextView txtItemName,txtOrderText,txtOrder;
        ImageView imgFood;
        public MyViewHolder(View itemView) {
            super(itemView);
            // get the reference of item view's
            txtItemName =  itemView.findViewById(R.id.txt_item_name);
            txtOrderText=  itemView.findViewById(R.id.txt_order_text);
            txtOrder =  itemView.findViewById(R.id.txt_order);
            }
    }
}

File: app/src/main/java/com/app/cheffyuser/cart/adapter/CustomOrderAdapter.kt
package com.app.cheffyuser.cart.adapter

import android.app.Activity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.app.cheffyuser.R
import com.app.cheffyuser.cart.models.CustomPlateResponseData
import com.app.cheffyuser.home.adapter.BaseViewHolder
import com.app.cheffyuser.home.adapter.RecyclerItemClickListener
import com.app.cheffyuser.utils.loadUrl

class CustomOrderAdapter(
    private val context: Activity,
    private val items: MutableList<CustomPlateResponseData?>?,
    private val clickListener: RecyclerItemClickListener
) : RecyclerView.Adapter<BaseViewHolder>() {
    private lateinit var myHolder: BaseViewHolder

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseViewHolder {
        val inflater = LayoutInflater.from(context)
        val view = inflater.inflate(R.layout.single_custom_order_row, parent, false)
        return ViewHolder(view, clickListener)
    }

    override fun onBindViewHolder(holder: BaseViewHolder, position: Int) {
        myHolder = holder
        holder.onBind(position)
    }


    override fun getItemCount(): Int {
        return items?.size ?: 0
    }

    fun add(item: CustomPlateResponseData) {
        items!!.add(item)
        notifyItemInserted(items.size - 1)
    }

    private fun remove(item: CustomPlateResponseData) {
        val position = items!!.indexOf(item)
        if (position > -1) {
            items.removeAt(position)
            notifyItemRemoved(position)
        }
    }

    internal inner class ViewHolder(
        itemView: View,
        private val clickListener: RecyclerItemClickListener
    ) : BaseViewHolder(itemView), View.OnClickListener {

        private var model: CustomPlateResponseData? = null

        //init views here
        private val foodimage = itemView.findViewById<ImageView>(R.id.img_food)
        private val foodname = itemView.findViewById<TextView>(R.id.txt_item_name)
        private val bids = itemView.findViewById<TextView>(R.id.txt_bids)
        private val txt_viewall = itemView.findViewById<TextView>(R.id.txt_viewall)


        override fun onBind(position: Int) {
            super.onBind(position)
            this.model = items!![position]

            if (!model?.customPlateImages.isNullOrEmpty())
                foodimage.loadUrl(model!!.customPlateImages!![0]!!.url)

            foodname.text = "${model!!.name}"

            val bidCount = model?.customPlateAuction?.bidCount
            if (bidCount == null) bids.text = "(0) Chef"
            else bids.text = "($bidCount) Chef"
        }

        init {
            txt_viewall.setOnClickListener(this)
            foodimage.setOnClickListener(this)
            foodname.setOnClickListener(this)
            bids.setOnClickListener(this)
            itemView.setOnClickListener(this)
        }

        override fun clear() {}

        override fun onClick(v: View?) {
            clickListener.modelClick(model!!)
        }
    }

    fun refreshList() {
        notifyDataSetChanged()
    }

    fun clearList() {
        val size = items!!.size
        items.clear()
        notifyItemRangeRemoved(0, size)
    }
}
